contrib/ci/check-headers.py

```diff
@@ -31,18 +34,46 @@ def test_files() -> int:
 
     toplevel_headers = ["libfwupd/fwupd.h", "libfwupdplugin/fwupdplugin.h"]
     toplevel_headers_nopath = [os.path.basename(fn) for fn in toplevel_headers]
+    lib_headers = glob.glob("libfwupd*/*.h")
+    lib_headers.remove("libfwupd/fwupd.h")
+    lib_headers.remove("libfwupdplugin/fwupdplugin.h")
+    lib_headers_nopath = [os.path.basename(fn) for fn in lib_headers]
+
+    # test all C and H files
+    for fn in glob.glob("**/*.[c|h]", recursive=True):
+        includes = __get_includes(fn)
+
+        if (
+            fn.startswith("plugins")
+            and not fn.endswith("self-test.c")
+            and not fn.endswith("-tool.c")
+        ):
+            for include in includes:
+                # check for using private header use in plugins
+                if include.endswith("private.h"):
+                    print("{} uses private header {}".format(fn, include))
+                    rc = 1
+                    continue
 
-    for toplevel_header in toplevel_headers:
+                # check for referring to anything but top level header
+                if include in lib_headers or include in lib_headers_nopath:
+                    print(
+                        "{} contains {}, should only use top level includes".format(
+                            fn, include
+                        )
+                    )
+                    rc = 1
 
-        toplevel_fn = os.path.basename(toplevel_header)
-        toplevel_includes = __get_includes(toplevel_header)
-        toplevel_includes_nopath = [os.path.basename(fn) for fn in toplevel_includes]
+        # check for double top level headers
+        for toplevel_header in toplevel_headers:
 
-        # test all C and H files
-        for fn in glob.glob("**/*.[c|h]", recursive=True):
-            includes = __get_includes(fn)
+            toplevel_fn = os.path.basename(toplevel_header)
+            toplevel_includes = __get_includes(toplevel_header)
+            toplevel_includes_nopath = [
+                os.path.basename(fn) for fn in toplevel_includes
+            ]
 
-            # we do not need both toplevel headers
+            # we do not need both toplevel haders
```

Comment:
you reverted my typo fix :)

Reply:
Yeah, I missed it when rebasing.
Fixed it.